### PR TITLE
Update kubebuilder binary version to 1.0.0

### PR DESCRIPTION
**Pull request by marun**
cc: @font

### CODE REVIEW

File: scripts/download-binaries.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ root_dir="$(cd "$(dirname "$0")/.." ; pwd)"
 dest_dir="${root_dir}/bin"
 mkdir -p "${dest_dir}"
 
-kb_version="0.1.12"
+kb_version="1.0.0"
```
Comment (font) [inline scripts/download-binaries.sh]: Does it include kube v1.11 binaries?

Comment (gyliu513) [reply scripts/download-binaries.sh]: I opened an issue at https://github.com/kubernetes-sigs/kubebuilder/issues/339 to request kubebuilder upgrade kubernetes to 1.11.

**Comment by irfanurrehman** [comment]
@marun, can you please update the version in the userguide also in this PR, as pointed out by @gyliu513 so that we can merge this. Thanks!

**Comment by marun** [comment]
Updated userguide reference as requested.

File: docs/userguide.md [hunk 2]
```diff
@@ -31,8 +31,8 @@ Or you can install them manually yourself using the guidelines provided below.
 
 This repo depends on
 [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder)
-to generate code and build binaries. Download the [v0.1.12
-release](https://github.com/kubernetes-sigs/kubebuilder/releases/tag/v0.1.12)
+to generate code and build binaries. Download the [v1.0.0
```
Comment (gyliu513) [inline docs/userguide.md]: image

**Comment by gyliu513** [comment]
@marun I think `hack/install-latest.yaml` should also be updated as the kubebuilder version changed. Shall we create another PR to handle this?

**Comment by marun** [comment]
@gyliu513 There is zero functional change to the yaml between 0.1.12 and 1.0.0, just the tagged version change.  I'd rather not update it until there is an actual need to do so.

**Comment by gyliu513** [comment]
@marun yes, only tag version change and does not impact function, but without the update, people may feel confused if they compare the `hack/install.yaml` generated by themselves with the `hack/install-latest.yaml` in fed v2 repo.

**Pull request merged**